Azure AD Reporting quarterly updates Q4 FY2021 (#13361)
microsoftgraph/microsoft-graph-docs: api-reference/beta/resources/signin.md

|isTenantRestricted|Boolean|Shows whether the sign-in event was subject to an Azure AD tenant restriction policy.|
|location|[signInLocation](signinlocation.md)|The city, state, and 2 letter country code from where the sign-in occurred. Supports `$filter` (`eq` and `startsWith` operators only) on **city**, **state**, and **countryOrRegion** properties.|
|networkLocationDetails|[networkLocationDetail](networklocationdetail.md) collection|The network location details including the type of network used and its names.|
|originalRequestId|String|The request identifier of the first request in the authentication sequence. Supports `$filter` (`eq` operator only).|
|privateLinkDetails|[privateLinkDetails](../resources/privatelinkdetails.md)|Contains information about the Azure AD Private Link policy associated with the sign-in event.|
|processingTimeInMilliseconds|Int|The request processing time in milliseconds in AD STS.|
|resourceDisplayName|String|The name of the resource that the user signed in to. Supports `$filter` (`eq` operator only).|
|resourceId|String|The identifier of the resource that the user signed in to. Supports `$filter` (`eq` operator only).|

> |resourceTenantId|String|The tenant identifier of the resource referenced in the sign in.|

Reviewer: ```suggestion
|resourceTenantId|String|The tenant identifier of the resource that the user signed in to.|
```

Author: Sometimes the sign in can be for a service principal, so referring to a user here is a little over specific.

Author: What if we said "The tenant identifier of the resource that the user or service principal signed in to"

Reviewer: Sounds good.

|riskLevelAggregated|riskLevel|The aggregated risk level. Possible values: `none`, `low`, `medium`, `high`, `hidden`, or `unknownFutureValue`. The value `hidden` means the user or sign-in was not enabled for Azure AD Identity Protection. Supports `$filter` (`eq` operator only). <br>**Note:** Details for this property are only available for Azure AD Premium P2 customers. All other customers are returned `hidden`.|
|riskLevelDuringSignIn|riskLevel|The risk level during sign-in. Possible values: `none`, `low`, `medium`, `high`, `hidden`, or `unknownFutureValue`. The value `hidden` means the user or sign-in was not enabled for Azure AD Identity Protection. Supports `$filter` (`eq` operator only). <br>**Note:** Details for this property are only available for Azure AD Premium P2 customers. All other customers are returned `hidden`.|
|riskState|riskState|The risk state of a risky user, sign-in, or a risk event. Possible values: `none`, `confirmedSafe`, `remediated`, `dismissed`, `atRisk`, `confirmedCompromised`, or `unknownFutureValue`. Supports `$filter` (`eq` operator only).|

> |servicePrincipalCredentialKeyId|String|The unique identifier of the key credential used by a service principal to authenticate in the sign-in event.|
> |servicePrincipalCredentialThumbprint|String|The certificate thumbprint of the certificate used by a service principal to authenticate in the sign-in event.|

Reviewer: ```suggestion
|servicePrincipalCredentialKeyId|String|The unique identifier of the key credential used by the service principal to authenticate.|
|servicePrincipalCredentialThumbprint|String|The certificate thumbprint of the certificate used by the service principal to authenticate.|
```

|servicePrincipalId|String|The application identifier used for sign-in. This field is populated when you are signing in using an application. Supports `$filter` (`eq` and `startsWith` operators only).|
|servicePrincipalName|String|The application name used for sign-in. This field is populated when you are signing in using an application. Supports `$filter` (`eq` and `startsWith` operators only).|

> |signInEventTypes|String collection|Indicates the category of sign in that the event represents. For user sign ins, the category can be `interactiveUser` or `nonInteractiveUser` and corresponds to the value for the **isInteractive** property on the signin resource. For managed identity sign ins, the category is `managedIdentity`. For service principal sign ins, the category is **servicePrincipal**. Possible values are: `interactiveUser`, `nonInteractiveUser`, `servicePrincipal`, `managedIdentity`, `unknownFutureValue`.|

Reviewer: ```suggestion
|signInEventTypes|String collection|Indicates the category of sign in that the event represents. For user sign ins, the category can be `interactiveUser` or `nonInteractiveUser` and corresponds to the value for the **isInteractive** property on the signin resource. For managed identity sign ins, the category is `managedIdentity`. For service principal sign ins, the category is **servicePrincipal**. Possible values are: `interactiveUser`, `nonInteractiveUser`, `servicePrincipal`, `managedIdentity`, `unknownFutureValue`. Supports `$filter` (`eq` operator only).|
```

> |signInEventTypes|String collection|Indicates the category of sign in the event represents. For user sign ins, the category can be interactiveUser or nonInteractiveUser; the category for a user sign in corresponds to the value for the isInteractive property on the signin resource. For managed identity sign ins, the category is managedIdentity. For service principal sign ins other than managed identity sign ins, the category is servicePrincipal. Possible values are: `interactiveUser`, `nonInteractiveUser`, `servicePrincipal`, `managedIdentity`, `unknownFutureValue`.|

Reviewer: ```suggestion
|signInEventTypes|String collection|Indicates the category of sign in that the event represents. For user sign ins, the category can be `interactiveUser` or `nonInteractiveUser` and corresponds to the value for the **isInteractive** property on the signin resource. For managed identity sign ins, the category is `managedIdentity`. For service principal sign ins, the category is **servicePrincipal**. Possible values are: `interactiveUser`, `nonInteractiveUser`, `servicePrincipal`, `managedIdentity`, `unknownFutureValue`.|
```

> |signInIdentifier|String|The identification a user provided to sign in - includes UPN, but is also populated when a user signs in using other identifiers.|
> |signInIdentifierType|signInIdentifierType|Provides the type of sign-in identifier. Possible values are: `userPrincipalName`, `phoneNumber`, `proxyAddress`, `qrCode`, `onPremisesUserPrincipalName`, `unknownFutureValue`.|

Reviewer: ```suggestion
|signInIdentifier|String|The identification that the user provided to sign in. It may be the userPrincipalName but it's also populated when a user signs in using other identifiers.|
|signInIdentifierType|signInIdentifierType|The type of sign in identifier. Possible values are: `userPrincipalName`, `phoneNumber`, `proxyAddress`, `qrCode`, `onPremisesUserPrincipalName`, `unknownFutureValue`.|
```

|status|[signInStatus](signinstatus.md)|The sign-in status. Includes the error code and description of the error (in case of a sign-in failure). Supports `$filter` (`eq` operator only) on **errorCode** property.|
|tokenIssuerName|String|The name of the identity provider. For example, `sts.microsoft.com`. Supports `$filter` (`eq` operator only).|
|tokenIssuerType|tokenIssuerType|The type of identity provider. Possible values: `AzureAD`, `ADFederationServices`, or `UnknownFutureValue`.|
|userAgent|String|The user agent information related to sign-in. Supports `$filter` (`eq` and `startsWith` operators only).|
|userDisplayName|String|The display name of the user. Supports `$filter` (`eq` and `startsWith` operators only).|
|userId|String|The identifier of the user. Supports `$filter` (`eq` operator only).|
|userPrincipalName|String|The UPN of the user. Supports `$filter` (`eq` and `startsWith` operators only).|

> |userType|signInUserType|Identifies whether the user is a member or guest in the tenant that performed the sign in. Possible values are: `member`, `guest`, `unknownFutureValue`.|

Reviewer: ```suggestion
|userType|signInUserType|Identifies whether the user is a member or guest in the tenant. Possible values are: `member`, `guest`, `unknownFutureValue`.|
```
